addons/udes_advanced_packaging/models/stock_quant_package.py

```diff
@@ -101,3 +101,34 @@ def write(self, vals):
         """
         self.validate_parent_name(vals)
         return super().write(vals)
+
+    def _get_all_products_quantities(self):
+        """This function computes the different product quantities for the given package"""
+        # TODO: Issue 962, make this work with different UoMs
+        res = {}
+        for quant in self._get_contained_quants():
+            if quant.product_id not in res:
+                res[quant.product_id] = 0
+            res[quant.product_id] += quant.quantity
+        return res
+
+    def assert_reserved_full_package(self, move_lines):
+        """Check that a package is fully reserved at move_lines."""
+        StockQuantPackage = self.env["stock.quant.package"]
+
+        self.ensure_one()
+
+        pack_products = frozenset(self._get_all_products_quantities().items())
+        mls_products = frozenset(move_lines._get_all_products_quantities().items())
+        if pack_products != mls_products:
+            # move_lines do not match the quants
+            picking = move_lines.picking_id
+            picking.ensure_one()
+            pack_mls = StockQuantPackage._get_current_move_lines()
+            other_pickings = pack_mls.picking_id - picking
+            if other_pickings:
+                raise ValidationError(
+                    _("The package is reserved in other pickings: %s")
+                    % ",".join(other_pickings.mapped("name"))
```

Comment:
is this .mapped() left because it's required or can it be removed?

Reply:
```➜  udes_advanced_packaging git:(story/459_v2) ag '.mapped'       
models/stock_picking.py
44:                                % ",".join(other_pickings.mapped("name"))

models/stock_quant_package.py
132:                    % ",".join(other_pickings.mapped("name"))
```

Reply:
I believe this is required, because it is not a relational field